### PR TITLE
Compute 3d power

### PR DESCRIPTION
**Pull request by qezlou**
- Computing 3D power using `nbodykit`
-  I haven't added `nbodykit` to the requirements because it is only needed in `fluxstatistics.py` and won't be needed for generating spectra.

### CODE REVIEW

**Comment by sbird** [comment]
Thanks Sum. Can you wrap the nbodykit imports in a 
try:
nbodykit...
except ImportError:
pass

so that fake_spectra is installable even if nbodykit isn't available?

**Comment by qezlou** [comment]
Thanks, That makes sense! Just sent a commit to this PR.

**Comment by sbird** [comment]
Also can you remove the binary file from the commit?

**Comment by qezlou** [comment]
> Also can you remove the binary file from the commit?

Sorry It slipped away. It is removed.

File: fake_spectra/fluxstatistics.py [hunk 1]
```diff
@@ -98,6 +106,92 @@ def flux_power(tau, vmax, spec_res = 8, mean_flux_desired=None, window=False):
...
+        We compute the power spectrum along each sightline and then average the result.
+        Arguments:
+        comm_nbodykit: MPI communicator for nbodykit, I prefer to have one communicator for each process, i.e.
+                        turning off parallel processing in nbodykit cause it is already fast enough. 
+                        You can set it as None if parallelism is not a concern to you.
+            tau - optical depths. Shape is (NumLos, npix)
+            mean_flux_desired - Mean flux to rescale to.
+            dk - bin width in k, units of h/cMpc
+            Nmu - number of mu bins, default is 10 which is usually used in 3D correlation function
+            quite - if you want to see ``nbodykit`'s loggings
```
Comment (sbird) [inline fake_spectra/fluxstatistics.py:142]: Can you spell this 'quiet'? :)

File: fake_spectra/fluxstatistics.py [hunk 2]
```diff
@@ -98,6 +106,92 @@ def flux_power(tau, vmax, spec_res = 8, mean_flux_desired=None, window=False):
...
+            mean_flux_desired = np.mean(np.exp(-tau))
+        (nspec, npix) = np.shape(tau)
+        nt = np.sqrt(nspec/3).astype(int)
+        x, y, z = np.meshgrid(np.arange(nt), np.arange(nt), np.arange(npix), indexing='ij')
+        x = x*boxsize/nt
+        y = y*boxsize/nt
+        z = z*boxsize/npix
+        coords = np.vstack((x.ravel(), y.ravel(), z.ravel())).T
+        for i in range(3):
+            if not quite:
```
Comment (sbird) [inline fake_spectra/fluxstatistics.py:164]: Here too

File: fake_spectra/fluxstatistics.py (hunk 1 above)
Comment (qezlou) [reply fake_spectra/fluxstatistics.py]: Haha, sorry, I just replaced all

**Comment by sbird** [comment]
Merged, thank you!

**Pull request merged**